test/digraph/transitivity.jl

```diff
@@ -27,4 +27,64 @@
         @test newcircle == @inferred(transitiveclosure!(circle, true))
         @test ne(circle) == 16
     end
+
+    # transitivereduction
+    let
+        nullgraph = SimpleDiGraph(0)
+        pathgraph = SimpleDiGraph(10)
+        pathgraph_closure = transitiveclosure(pathgraph)
+        complete_graph = CompleteDiGraph(9)
+        complete_graph_reduced = transitivereduction(complete_graph)
+        no_edge_graph = SimpleDiGraph(7)
+        self_loop_graph = SimpleDiGraph(1)
+        add_edge!(self_loop_graph, 1, 1)
+        
+        barbell_graph = SimpleDiGraph(9)
+        for i in 1:4, j in 1:4
+            i == j && continue
+            add_edge!(barbell_graph, i, j)
+        end
+        barbell_graph = SimpleDiGraph(9)
+        for i in 1:4, j in 1:4
+            i == j && continue
+            add_edge!(barbell_graph, i, j)
+        end
+        for i in 5:9, j in 5:9
+            i == j && continue
+            add_edge!(barbell_graph, i, j)
+        end
+        add_edge!(barbell_graph, 1,5);
+
+        # transitive reduction of a the nullgraph is again a nullgraph
+        @test nullgraph == transitivereduction(nullgraph)
```

Comment:
These should be tested on graphs of varying `eltype`s: that is, see the `testgraphs` / `testdigraphs` functions:
```
for g in testgraphs(barbell_graph)
   ... do tests
end
```

Reply:
Is the idea that we use testdigraphs for all testcases or should I just do this for this barbell_graph? I might have to replace that test anyway as I have to come up with another example that covers all lines.

Reply:
In general, we prefer testgraphs / testdigraphs for all test cases, inside the loop.